### ISSUE TITLE
ChartWidthMixin for es6/es7 class?

### ISSUE THREAD

**Issue by raptoria** [rrag/react-stockcharts#32]
Any workaround for using the ChartWidthMixin in es6/es7 classes? Do you recommend say, https://www.npmjs.com/package/react-mixin-decorator ?

**Comment by rrag** [comment]
Disclaimer: I have not tried this, but it should work, this is similar to [wrap.jsx](https://github.com/rrag/react-stockcharts/blob/master/src/lib/series/wrap.jsx) and [pure.jsx](https://github.com/rrag/react-stockcharts/blob/master/src/lib/pure.jsx)

the logic in the `ChartWidthMixin` is quite simple, I think you can create a function something like

```js
export default function makeResponsive(Component) {
	return class ResponsiveComponent extends React.Component {
		componentDidMount() {
			window.addEventListener("resize", this.handleWindowResize);
			var el = React.findDOMNode(this);
			var w = el.parentNode.clientWidth;
			this.setState({
				width: w
			});
		}
		componentWillUnmount() {
			window.removeEventListener("resize", this.handleWindowResize);
		}
		handleWindowResize() {
			var el = React.findDOMNode(this);
			var w = el.parentNode.clientWidth;
			this.setState({
				width: w
			});
		}
		render() {
			if (this.state && this.state.width)
				return <Component width={this.state.width} {...this.props} />
			else
				return <div />
		}
	}
}
```

and in your `CandleStickChart.jsx`

```js
class CandleStickChart extends React.Component {
	...
	...
}

export default makeResponsive(CandleStickChart);
```

**Comment by raptoria** [comment]
thanks for the prompt response! I edited it slightly. there was a small error on the window resize event (it wasn't getting the 'this' that we want).


       function withResponsive(ComposedComponent) {
                  return class withResponsive extends Component {

                      componentDidMount() {
                            this.resizeListener = () => this.handleWindowResize();
                            window.addEventListener("resize", this.resizeListener );
                            var el = React.findDOMNode(this);
                            var w = el.parentNode.clientWidth;
                            this.setState({
                                width: w
                            });
                        }
                        componentWillUnmount() {
                            window.removeEventListener("resize", this.resizeListener );
                        }
                        handleWindowResize() {
                            var el = React.findDOMNode(this);
                            var w = el.parentNode.clientWidth;
                            this.setState({
                                width: w
                            });
                        }
                        render() {
                            if (this.state && this.state.width)
                                return <ComposedComponent width={this.state.width} {...this.props} />
                            else
                                return <div />
                        }


                  };
                }

                export default withResponsive;




then to use it inside Histogram or CandleStickChart, you need only add:

@withResponsive

and under PropTypes,
     width: React.PropTypes.number.isRequired,

**Comment by rrag** [comment]
ah, I forgot to bind this. It was close though :)

I suggest not using 

```js
this.resizeListener = () => this.handleWindowResize();
```

but instead add
```js
constructor(props) {
   super(props);
   this.handleWindowResize = this.handleWindowResize.bind(this);
}
```

**Comment by rrag** [comment]
thanks for asking, updated all the examples to use es6 classes, 

http://plnkr.co/edit/gist:a8465abe0061df1b7976?p=preview

look for `fitWidth`

Have not published this to npm yet, hopefully next week.

**Comment by raptoria** [comment]
nice, that's much cleaner.

thanks!!

**Comment by rrag** [comment]
published v0.2.11
